modules/sfp_keybase.py

```diff
@@ -0,0 +1,241 @@
+
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+import urllib.request, urllib.parse, urllib.error
+import json
+import re
+
+class sfp_keybase(SpiderFootPlugin):
+    """Keybase:Footprint,Investigate,Passive:Public Registries::Obtain additional information about target username"""
+
+    opts = {
+    }
+
+    optdescs = {
+    }
+
+    # Tracking results can be helpful to avoid reporting/processing duplicates
+    results = None
+
+    # Tracking the error state of the module can be useful to detect when a third party
+    # has failed and you don't wish to process any more events.
+    errorState = False
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    def watchedEvents(self):
+        return ["USERNAME", "LINKED_URL_EXTERNAL"]
+
+    def producedEvents(self):
+        return ["RAW_RIR_DATA", "SOCIAL_MEDIA", "USERNAME",
+            "HUMAN_NAME", "GEOINFO", "BITCOIN_ADDRESS", 
+            "PGP_KEY"]
+
+    def queryUsername(self, qry):
+
+        params = {
+            'usernames': qry.encode('raw_unicode_escape').decode("ascii", errors='replace')
+        }
+
+        headers = {
+            'Accept' : "application/json"
+        }
+
+        res = self.sf.fetchUrl(
+          'https://keybase.io/_/api/1.0/user/lookup.json?' + urllib.parse.urlencode(params),
+          headers=headers,
+          timeout=15,
+          useragent=self.opts['_useragent']
+        )
+
+        # In this case, it will always be 200 if keybase is queried 
+        # The actual response codes are stored in status tag of the response
+        if not res['code'] == '200':
+            return None
+        
+        content = json.loads(res['content'])
+
+        status = content.get('status')
+        if status is None:
+            return None
+
+        code = status.get('code')
+        if code is None:
+            return None
+        
+        try:
+            if not int(code) == 0:
+                return None
+        except:
+            self.sf.error("Invalid code returned as response", False)
+            return None
+
+        return content
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        if self.errorState:
+            return None
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+
+        self.results[eventData] = True
+
+        userName = eventData
+
+        # Extract username if a Keybase link is received 
+        if eventName == "LINKED_URL_EXTERNAL":
+            linkRegex = "keybase.io\/[A-Za-z0-9\-_\.]+"  
```

Comment:
May as well include `https?://` before `keybase.io` so you don't pick up external links with keybase.io, e.g. `https://someotherdomain.com/somepage.html?somevar=keybase.io/abc123`